### DISCUSSION TITLE
Why are the apt sources removed from /etc/apt/sources.list.d/ ?

### DISCUSSION THREAD

**Discussion by michaelCTS** [actions/virtual-environments#8241] [General questions]
This impacts things like installing google cloud SDK components.

`apt-get update && apt-get install -yqq google-cloud-cli-datastore-emulator` fails because the package doesn't exist as the sources were deleted

https://github.com/actions/runner-images/blob/ubuntu22/20230821.1/images/linux/scripts/installers/google-cloud-sdk.sh#L16C4-L17

The repo url is then stored in a file https://github.com/actions/runner-images/blob/ubuntu22/20230821.1/images/linux/scripts/installers/google-cloud-sdk.sh#L20 

that doesn't seem to be used only written to (according to [github's search](https://github.com/search?q=repo%3Aactions%2Frunner-images%20apt-sources.txt&type=code).

What gives?

**Comment by v-sergei-pyshnoi** [comment]
Hello @michaelCTS .During the image generation process, several sources are added to the apt repositories list. The main reason for removing them afterwards is to avoid various conflicts when installing/updating subsequent software.

**Reply by michaelCTS** [reply]
Are there recommendations for updating the packages within those repositories when running the image? 

The google cloud SDK is an example where being managed by the package manager makes it impossible to install additional components. This often leads to rewriting the logic involved in adding the package in the first place. gcloud is not an outlier here as other packages require updating sometimes too, leading the the same effort on the developer's side.

**Reply by vpolikarpov-akvelon** [reply]
@michaelCTS, if you need package source, you can add it again in your workflow. Like this:
```
echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
wget -qO- https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor | sudo tee -a /usr/share/keyrings/cloud.google.gpg
```
